powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/generator/HashBasedCounterGenerator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.security.powerauth.crypto.lib.generator;
+
+import io.getlime.security.powerauth.crypto.lib.util.Hash;
+
+/**
+ * Generator of hash based counter.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ *
+ */
+public class HashBasedCounterGenerator {
```

Comment:
I would consider simplifying the naming here a bit:

```java
HashBasedCounter ctr = new HashBasedCounter(); // implements Counter...
byte[] seed = ctr.init();
byte[] next = ctr.next(seed);
```

The idea is that this is a very simple internal class and we would like to emulate this code as simple as possible:

```java
int i = 0;
i++;
```

Reply:
Yes, it's more concise, let's use it.